src/server/cluster/cluster_family.cc

```diff
@@ -43,6 +45,8 @@ constexpr string_view kClusterDisabled =
     "Cluster is disabled. Enabled via passing --cluster_mode=emulated|yes";
 constexpr string_view kDflyClusterCmdPort = "DflyCluster command allowed only under admin port";
 
+thread_local unique_ptr<ClusterConfig> tl_cluster_config;
```

Comment:
how about using shard_ptr?

Reply:
`shared_ptr` has the penalty of atomic operations on every access, so it kind of beats the purpose of thread local.

Also, it doesn't protect the inner object, which must remain thread safe if accessed by multiple threads, so we'll have to reintroduce the mutex inside ClusterConfig.

Reply:
I think that the atomic operations are in the control group of the shared ptr and when calling the get function there is not atomic operation. Therefor on the main flow of queries we will call the get function we will have no overhead.
If we will use the reset function we will not need to protect the inner object, and the atomic operations on the control group (changing the ref count) will happen only when we have the cluster config command running.

Reply:
You are right.
I thought that your proposal was to use a `shared_ptr` without `thread_local`. But combining the two is a good call.
https://github.com/dragonflydb/dragonfly/pull/1395

Reply:
Btw, I just want to add something about using atomics. (not related to this PR). Atomics do not impose huge penalty by themselves - yes they are of course less efficient than writing to memory due to L1,L2 caches etc but it's not end of the world. Where they become super slow (and I can demonstrate cases of having 1000  slowdown) is when they are contended. 
so 64 threads writing into 64 different atomics - it's okeyish. 64 threads writing into a single atomic - that's a show-stopper that will bring down your server performance to be slower than a single-threaded system!